openstack/sharedfilesystems/v2/shares/results.go

```diff
@@ -76,6 +77,20 @@ func (r commonResult) Extract() (*Share, error) {
 	return s.Share, err
 }
 
+// SharePage is a pagination.pager that is returned from a call to the List func
+type SharePage struct {
+	pagination.SinglePageBase
```

Comment:
Based on the discussion in #134 I'm starting to question if `SingleBasePage` should be a `MarkerBasePage`. Given that in the Manila API we support `limit` and `offset`, maybe it should be? 

I have a naive acceptance test for pagination calling `EachPage()` for shares, which seems to work with `SinglePageBase`, but I'm not confident that the test would reveal a problem.